numpy/lib/tests/test_index_tricks.py

```diff
@@ -175,6 +175,24 @@ def test_0d(self):
         assert_raises_regex(
             ValueError, "out of bounds", np.unravel_index, [1], ())
 
+    @pytest.mark.parametrize("mode", ["clip", "wrap", "raise"])
+    def test_empty_array_ravel(self, mode):
+        res = np.ravel_multi_index(
+                    np.zeros((3, 0), dtype=np.intp), (2, 1, 0), mode=mode)
+        assert(res.shape == (0,))
+
+        with assert_raises(ValueError):
+            np.ravel_multi_index(
+                    np.zeros((3, 1), dtype=np.intp), (2, 1, 0), mode=mode)
```

Comment:
Does this change the error message for `mode="raise"`?

Reply:
The original error (non empty) is: "invalid entry in coordinates array", which is fair. Could just use that, neither is great to be honest.